packages/directlinespeech/src/createWebSpeechPonyfillFactory.js

```diff
@@ -0,0 +1,90 @@
+/* eslint class-methods-use-this: ["error", { "exceptMethods": ["cancel", "getVoices", "speak"] }] */
+
+import { AudioConfig } from 'microsoft-cognitiveservices-speech-sdk';
+import { createSpeechRecognitionPonyfillFromRecognizer } from 'web-speech-cognitive-services/lib/SpeechServices/SpeechToText';
+import AbortController from 'abort-controller';
+
+import createErrorEvent from './createErrorEvent';
+import createTaskQueue from './createTaskQueue';
+import EventTargetShim, { defineEventAttribute } from './external/event-target-shim';
+import playCognitiveServicesStream from './playCognitiveServicesStream';
+import playWhiteNoise from './playWhiteNoise';
+import SpeechSynthesisAudioStreamUtterance from './SpeechSynthesisAudioStreamUtterance';
+
+export default function({
+  audioConfig = AudioConfig.fromDefaultMicrophoneInput(),
+  enableTelemetry,
+  recognizer,
+  textNormalization
+}) {
+  return () => {
+    const { SpeechGrammarList, SpeechRecognition } = createSpeechRecognitionPonyfillFromRecognizer({
+      audioConfig,
+      createRecognizer: () => recognizer,
+      enableTelemetry,
+      looseEvents: true,
+      textNormalization
+    });
+
+    const audioContext = new AudioContext();
+
+    const { cancelAll, push } = createTaskQueue();
+
+    class SpeechSynthesis extends EventTargetShim {
+      cancel() {
+        cancelAll();
+      }
+
+      getVoices() {
+        return [];
```

Comment:
why is this array empty? Shouldn't it return the voices, if they exist?

Reply:
Agreed! To me, it's reasonable to return an empty array, but the array shouldn't be defined inline like this, it should be necessary and used elsewhere inside the class. `return this.voices`

Reply:
1. We cannot detect what type of voice is supported;
2. Synthesis is done on the bot side, thus, these voices should not be used.